core/src/main/java/org/jruby/runtime/backtrace/TraceType.java

```diff
@@ -581,6 +602,19 @@ public static IRubyObject generateMRIBacktrace(Ruby runtime, RubyStackTraceEleme
         return RubyArray.newArrayMayCopy(runtime, traceArray);
     }
 
+    public static String maskInternalFiles(String filename) {
+        if (isInternal(filename)) {
+            return "<internal:" + filename + ">";
+        }
+
+        return filename;
+    }
+
+    public static boolean isInternal(String filename) {
+        return filename.startsWith("uri:classloader:/jruby/kernel/") ||
+                filename.startsWith("<internal:");
```

Comment:
seems like the above could than theoretically double mask as `"<internal:<internal:" + ...` if the check is present in `maskInternalFiles` and a masked `filename` gets passed...

Reply:
You're right... I think this code evolved as I went and became general "is this source file internal" check. I will look into it and figure out why I did this.

Reply:
@kares I believe my last commit fixes and clarifies this logic.